backpack-compose/src/main/kotlin/net/skyscanner/backpack/compose/navigationbar/internal/BpkTopNavBarImpl.kt

```diff
@@ -60,10 +61,12 @@ internal fun BpkTopNavBarImpl(
     insets: WindowInsets? = null,
     navIcon: IconAction? = null,
     actions: List<Action> = emptyList(),
+    transparent: Boolean = false,
 ) {
+    val backgroudColor = if (fraction == 0f) BpkTheme.colors.surfaceDefault else (if (transparent) Color.Transparent else BpkTheme.colors.canvas)
     Surface(
-        color = animateColorAsState(targetValue = if (fraction == 0f) BpkTheme.colors.surfaceDefault else BpkTheme.colors.canvas).value,
-        contentColor = BpkTheme.colors.textPrimary,
+        color = animateColorAsState(targetValue = backgroudColor).value,
+        contentColor = if (transparent && fraction > 0f) BpkTheme.colors.textOnDark else BpkTheme.colors.textPrimary,
```

Comment:
I believe it's always `textOnDark` if transparent - what's the `fraction > 0f` check for?
also possibly worth extracting to a separate property as well, like the `backgroundColor`

Reply:
@marianeum  It seems that fraction < 0f indicates the expanded status. The textOnDark color should be applied only in expanded status rather than collapsed status.